binance/threaded_stream.py

```diff
@@ -53,8 +53,11 @@ async def start_listener(self, socket, path: str, callback):
                     continue
                 else:
                     if not msg:
-                        continue
-                    callback(msg)
+                        continue # Handle both async and sync callbacks
+                    if asyncio.iscoroutinefunction(callback):
+                        await callback(msg)
```

Comment:
I tried to make this non blocking by using `self._loop.call_soon_threadsafe()` but kept running into different errors